fabrics.c

```diff
@@ -1107,11 +1106,17 @@ int nvmf_disconnect(const char *desc, int argc, char **argv)
 	nvme_root_skip_namespaces(r);
 	ret = nvme_scan_topology(r, NULL, NULL);
 	if (ret < 0) {
-		if (errno != ENOENT)
-			fprintf(stderr, "Failed to scan topology: %s\n",
-				nvme_strerror(errno));
-		nvme_free_tree(r);
-		return ret;
+		/*
+		 * Do not report an error when the modules are not
+		 * loaded, this allows the user to uncodinitally call
+		 * disconnect.
+		 */
+		if (errno == ENOENT)
+			return 0;
```

Comment:
This looks great for `nvme disconnect-all` and `nvme disconnect --nqn`.
However, for `nvme disconnect --device`, perhaps we should report an error if the module is unloaded? Because whatever device(s) the user is requesting to disconnect definitely don't exist.

Reply:
> Because whatever device(s) the user is requesting to disconnect definitely don't exist.

One could argue that if a user tries to disconnect a device that doesn't exist, then the operation is a success. In the end, the user is really trying to remove the device. The end result is achieved whether the device existed or not when the disconnect was issued.

Reply:
Fair. I am fine with it this way. It's certainly simpler to handle all the disconnect commands the same way.

Reply:
This was also my thinking. In the end, the user wanted that the device is disconnected and it is.